WooCommerce/Classes/ViewRelated/MainTabBarController.swift

```diff
@@ -698,12 +727,12 @@ extension MainTabBarController: DeepLinkNavigator {
 // MARK: - Site ID observation for updating tab view controllers
 //
 private extension MainTabBarController {
-    func observePOSEligibilityForPOSTabVisibility(siteID: Int64) {
-        guard let posTabVisibilityChecker else {
-            updateTabViewControllers(isPOSTabVisible: false, isBookingsTabVisible: isBookingsTabVisible)
-            viewModel.loadHubMenuTabBadge()
-            return
-        }
+    func observePOSEligibilityForPOSTabVisibility(site: Site) {
+        let siteID = site.siteID
+
+        // Configures POS tab coordinator once per logged in site session.
+        let posTabVisibilityChecker = posTabVisibilityCheckerFactory(site)
+        self.posTabVisibilityChecker = posTabVisibilityCheckerFactory(site)
```

Comment:
Nit: why don't we set the checker using the local variable?


```suggestion
        self.posTabVisibilityChecker = posTabVisibilityChecker
```

Reply:
Done in [dec1f90](https://github.com/woocommerce/woocommerce-ios/pull/16292/commits/dec1f9087a54b9de37e58307059a130fc08792f3)